sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnInvokers.java

```diff
@@ -184,35 +176,32 @@ public String subclass(TypeDescription.Generic superClass) {
     return res;
   }
 
-  /** Implements an invoker method by doing nothing and immediately returning void. */
-  private static class NoopMethodImplementation implements Implementation {
-    @Override
-    public InstrumentedType prepare(InstrumentedType instrumentedType) {
-      return instrumentedType;
-    }
+  /** Delegates to the given method if available, or does nothing. */
+  private static Implementation delegateOrNoop(DoFnSignature.DoFnMethod method) {
+    return (method == null)
+        ? FixedValue.originType()
+        : new SimpleMethodDelegation(method.targetMethod());
```

Comment:
If I'm not mistaken, won't this turn all delegations into `SimpleMethodDelegation`? Does that now subsume the functionality of `BundleMethodDelegation` and `LifecycleMethodDelegation`?